core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -468,11 +468,18 @@ private[spark] class TaskSchedulerImpl(
         removeExecutor(executorId, reason)
         failedExecutor = Some(executorId)
       } else {
-         // We may get multiple executorLost() calls with different loss reasons. For example, one
-         // may be triggered by a dropped connection from the slave while another may be a report
-         // of executor termination from Mesos. We produce log messages for both so we eventually
-         // report the termination reason.
-         logError("Lost an executor " + executorId + " (already removed): " + reason)
+         executorIdToHost.get(executorId) match {
+           case Some(_) =>
+             // We may need to update rootPool in case the executor was pending a real loss reason.
```

Comment:
What about changing this comment to something like:
"If the executor is still in executorIdToHost, it means we haven't gotten the real loss reason yet, so still call removeExecutor to fail all of the tasks running on that executor (with the real loss reason)." (as-is, it's not clear what it means to update rootPool)